packages/engine/src/credentials/CredentialSystem.ts

```diff
@@ -0,0 +1,73 @@
+import { NotificationService } from '@xrengine/client-core/src/common/services/NotificationService'
+import { requestVcForEvent, vpRequestQuery } from '@xrengine/common/src/credentials/credentials'
+import { CallbackComponent } from '@xrengine/engine/src/scene/components/CallbackComponent'
+import { createActionQueue, dispatchAction, getState } from '@xrengine/hyperflux'
+
+import { isClient } from '../common/functions/isClient'
+import { Engine } from '../ecs/classes/Engine'
+import { addComponent, defineQuery, getComponent } from '../ecs/functions/ComponentFunctions'
+import { NameComponent } from '../scene/components/NameComponent'
+import { CredentialAction } from './CredentialAction'
+import { CredentialState } from './CredentialState'
+
+function updateCredentialState(action: typeof CredentialAction.verificationResult.matches._TYPE) {
+  console.log('in updateCredentialState')
+  if (action.$from !== Engine.instance.currentWorld.worldNetwork.hostId) return
+  const credential = action.credential
+  const credentialState = getState(CredentialState)
+  if (action.verified) {
+    // dispatch success notification
+    console.log('Verification successful!')
+    NotificationService.dispatchNotify('Verification successful!', { variant: 'success' })
+  } else {
+    // dispatch error notification
+    console.log('Verification fail!')
+    NotificationService.dispatchNotify('Verification failed!', { variant: 'error' })
+  }
+  if (!credentialState[action.userId].value) credentialState[action.userId].set({ all: [] })
+  credentialState[action.userId].all.merge(credential)
+}
+
+export const CredentialSystem = () => {
+  const didVerifyCredentialActions = createActionQueue(CredentialAction.verificationResult.matches)
+  const nameQuery = defineQuery([NameComponent])
+
+  return () => {
+    for (const entity of nameQuery.enter()) {
+      const name = getComponent(entity, NameComponent).name
+      if (name === 'Door' || name === 'VCKey Trigger Volume') {
```

Comment:
we can't rely on component names here, we should be creating a tag component, and use that to identify an object that is requiring this functionality

Reply:
@HexaField yeah, this was a temporary hack me and Gheric used for the demo -- we figured we'd discuss with you re how to do it properly

Reply:
this still needs to be addressed

Reply:
Future PR.